ts/components/messages/MessageDueDateBar.tsx

```diff
@@ -57,182 +70,210 @@ const styles = StyleSheet.create({
+    : status === "expiredAndExpirable"
+    ? { backgroundColor: customVariables.brandDarkGray }
+    : status === "expiredNotExpirable"
+    ? { backgroundColor: customVariables.calendarExpirableColor }
+    : status === "expiring"
+    ? { backgroundColor: customVariables.calendarExpirableColor }
+    : status === "valid"
+    ? { backgroundColor: customVariables.brandGray }
+    : assertNever(status);
+
+const isPaymentExpirable = (message: CreatedMessageWithContent): boolean =>
+  paymentExpirationInfo(message).fold(false, isExpirable);
+
+const isPaymentExpired = (message: CreatedMessageWithContent): boolean =>
+  paymentExpirationInfo(message).fold(false, isExpired);
+
+const isPaymentExpiring = (message: CreatedMessageWithContent): boolean =>
+  paymentExpirationInfo(message).fold(false, isExpiring);
+
+const paid = (payment: PaidReason | undefined): boolean =>
+  payment !== undefined;
+
+const calculatePaymentStatus = (
+  payment: PaidReason | undefined,
+  message: CreatedMessageWithContent
+): PaymentStatus =>
+  isPaymentExpired(message) && isPaymentExpirable(message)
+    ? "expiredAndExpirable"
+    : isPaymentExpired(message) && !isPaymentExpirable(message)
+    ? "expiredNotExpirable"
+    : isPaymentExpiring(message)
+    ? "expiring"
+    : paid(payment)
+    ? "paid"
+    : "valid";
+
+/**
+ * A component to show detailed info about the due date of a message
+ */
+const MessageDueDateBar: React.FunctionComponent<Props> = ({
+  onGoToWallet,
+  message,
+  payment
+}) => {
   /**
    * Display description on message deadlines
    */
-  public render() {
-    const { dueDate, paid } = this;
-
-    if (dueDate.isNone()) {
-      return null;
-    }
-
-    return (
-      <React.Fragment>
-        <View
-          style={[
-            styles.container,
-            this.bannerStyle,
-            paid ? styles.center : undefined
-          ]}
-        >
-          <React.Fragment>
-            {this.renderCalendarIcon()}
-            <View hspacer={true} small={true} />
-            <Text
-              style={styles.text}
-              white={
-                !this.paid && (this.isPaymentExpiring || this.isPaymentExpired)
-              }
-            >
-              {this.textContent}
-            </Text>
-          </React.Fragment>
-        </View>
-        <View spacer={true} large={true} />
-      </React.Fragment>
-    );
-  }
-}
+
+  const paymentStatus = calculatePaymentStatus(payment, message);
+  return fromNullable(message.content.due_date).fold(null, dueDate => (
+    <>
+      <View
+        style={[
+          styles.container,
+          bannerStyle(paymentStatus),
+          paymentStatus === "paid" ? styles.center : undefined
+        ]}
+      >
+        <>
+          <CalendarIcon status={paymentStatus} dueDate={dueDate} />
+          <View hspacer={true} small={true} />
+
+          <Text
+            style={styles.text}
+            white={
+              paymentStatus === "expiring" ||
```

Comment:
maybe a function `isExpiringOrExpired():boolean` could be a good refactoring?

Reply:
In this case I don't think it's greater value, the condition is quite eloquent as the dirty job has been done reducing the state to a scalar `PaymentStatus` value

Reply:
If the goal is to have `white` for _expired_/_expiring_ payments, I guess `isExpiringOrExpired` function could be more expressive than 3 conditions in `or`
I knew the point but a new developer has to know what `expiring`, `expiredAndExpirable`, `expiredNotExpirable`
in `or` means.

imho it reduces the cognitive effort, btw it's the same and this is only my pov ;)

Reply:
done